### ISSUE TITLE
Feature Request: Support for Self Hosted image manipulation server (Thumbor, Imgproxy, Imaginary)

### ISSUE THREAD

**Issue by imsus** [ascorbic/unpic#37]
Please support the open source alternatives

### Thumbor
Website: https://www.thumbor.org/
Github Repo: https://github.com/thumbor/thumbor

### Imgproxy
Website: https://imgproxy.net/
Github Repo: https://github.com/imgproxy/imgproxy

### Imaginary
Github Repo: https://github.com/h2non/imaginary

**Comment by ascorbic** [comment]
Yeah, ipx too. What I want to do is introduce a more concept of a fallback transformer, which only handles transforms if it doesn't match a CDN transformer. I can take a lot of the logic out of the Vercel transformer. That would make it a lot easier to implement these.

**Comment by lilith** [comment]
I'd also like to suggest my self-hosted OSS image servers Imageflow and ImageResizer; both have the same querystring command syntax. 

https://docs.imageflow.io/querystring/introduction.html
https://github.com/riapi/riapi

**Comment by rajeshg** [comment]
+1 for imaginary. 

Here is their API
https://github.com/h2non/imaginary#http-api

**Comment by kdevan** [comment]
I'm thinking about writing a transformer for Imaginary. I've looked at some of the other transformers to get a feel for it. I wonder if you have any advice for how this process might look, since it will be using a custom domain and then going from unpic -> unpic-img which is my goal here.

**Comment by ascorbic** [comment]
The trouble with most of these is that they use signed URLs, which we can't support

**Comment by sahidrahman404** [comment]
> The trouble with most of these is that they use signed URLs, which we can't support

can we expose a function to fetch signed URLs from the backend

**Comment by ascorbic** [comment]
You'd not be able to do that if you ever wanted to use this in the client, e.g. if you're using unpic-img

**Comment by ascorbic** [comment]
I have implemented support for ipx. If anyone would like to add support for other self-hosted servers then I would suggest using that as an example

**Comment by anthonyholmes** [comment]
Would you be open to adding support for imgproxy with unsigned URLs? It seems like setting the allowed sources can act as security measure and unpic could add the disclaimer it only works with unsigned URLs
https://docs.imgproxy.net/configuration/options#IMGPROXY_ALLOWED_SOURCES

**Comment by ascorbic** [comment]
Absolutely! Please do. I'd need an example host for the tests though.

**Comment by anthonyholmes** [comment]
I can send you a couple links from my server unless you think that's a terrible idea

**Comment by ascorbic** [comment]
That should be fine. You can add that as part of the PR if that's ok

**Comment by ascorbic** [comment]
@turnah do they have any examples in the imgproxy docs, or anything like that?

**Comment by turnah** [comment]
> [@turnah](https://github.com/turnah) do they have any examples in the imgproxy docs, or anything like that?

They do, but unfortunately they have a signature so would need the salt and key to make any changes to the url meaning they're off limits (https://docs.imgproxy.net/usage/signing_url)

**Comment by ascorbic** [comment]
Incidentally, now that Unpic supports provider options, it _could_ support signed URLs. It would of course still need a key.
